### DISCUSSION TITLE
How to avoid duplicated published messages

### DISCUSSION THREAD

**Discussion by kevincaradant** [Koenkk/zigbee2mqtt#22358] [General]
Hi everyone,

I'm experiencing something I would like avoid. Publish twice the same message. 

![image](https://github.com/Koenkk/zigbee2mqtt/assets/6093175/3c3df328-3a41-4a22-bc43-ab5508b49123)

I've filtered some fields to alleviate Z2M/the coordinator and also HA (DB). However, I'm still receiving identical messages.

Have you found a way to realize it ? 

Thank you.

Edit: I read something around of filtered_cache option. But not sure to understand how it works:
filtered_cache Allows preventing certain attributes from ending up in the cache. This prevents attributes from being published when the value did not change.

It didn't work as expected because instead of not published the message, it published it with 'null' as value when it's the same value as before so it's not that I want :/
Cf:
![image](https://github.com/Koenkk/zigbee2mqtt/assets/6093175/c5d4c74d-fe2f-4015-b82a-1f471d5af608)

Thank you for your advices

**Comment by jokakilla** [comment]
I'm also doing some fine-tuning on my z2m installation and have some devices where I for example only care about state (on/off). With filtered_attributes it was possible to remove all the other attributes from the published message but like you I see lots of updates where nothing has really changed.

Looking at controller.ts in the source it seems like filtered_attributes doesn't have any impact on the things that cause a message to be published. The decision is made and the filtering is just done before sending out the publish.

I'm not so familiar with the code but was hoping that maybe the filtered_cache could help here. But it didn't change anything.

Did you find a solution?

______
Edit: At least from just quickly scanning the code there might be no "change" check before publishing. [publishEntityState](https://github.com/Koenkk/zigbee2mqtt/blob/master/lib/controller.ts#L325) Before mqtt.publish the message would need to be compared to the old messag I guess. In detail it might not be easy.

**Reply by jokakilla** [reply]
The solution is possibly much easier: Set the reporting Interval for Attributes you don't need very high. 65535 should disable reporting. No report -> No publish trigger.

I'll try that soonish.
